pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentRestletResource.java

```diff
@@ -230,19 +232,32 @@ protected Representation toggleSegmentState(String tableName, String segmentName
 
     JSONArray ret = new JSONArray();
     String tableNameWithType = null;
-    if ((tableType == null || TableType.OFFLINE.name().equalsIgnoreCase(tableType))
+    List<String> segmentsToToggle;
+
+    if (segmentName != null) {
+      segmentsToToggle = Collections.singletonList(segmentName);
+    } else {
+      segmentsToToggle = _pinotHelixResourceManager.getAllSegmentsForResource(tableName);
```

Comment:
the call getAllSegmentsForRersource() needs a table type appended correctly to the resource name. So, this call will not work